Automattic/wp-calypso
# Tailored flows: Implement random patterns in LiB

#### Proposed Changes

* This imports Calypso's persisted `QueryClient` to be able to persist the design order cleanly.
* It then preserves the order using a `useQuery` hook.

#### Testing Instructions
1. Go to /setup/link-in-bio/patterns.
2. The first four designs should be random.
3. Refresh the page, they should have the same random order. 
4. Repeat incognito. Should get another order.
5. Refresh the page, they should have the same random order. 

#### Risk
This has the potential to change every query behavior in Stepper. From what I see, there is only one risky hook, which is this [one](https://github.com/Automattic/wp-calypso/blob/add/randomize-patterns-in-lib/client/landing/stepper/hooks/use-theme-details.ts#L15) written by @markbiek. Hi Mark! 👋🏼 Do you think it would be OK if this hook actually had infinite `staleTime`? If not, please let me know and I'll make it shorter/fetch afresh on every load.
